### PR TITLE
Fixed code of Smartnest and updated documentation

### PR DESCRIPTION
**Pull request by DevilPro1**

### CODE REVIEW

File: usermods/smartnest/usermod_smartnest.h [hunk 1]
```diff
@@ -168,4 +168,40 @@ class Smartnest : public Usermod
...
     return USERMOD_ID_SMARTNEST;
   }
+
+  /**
+   * setup() is called once at startup to initialize the usermod.
+   */
+  void setup() {
+    // Initialization code here
+    if (!initialized) {
+      Serial.begin(115200);
```
Comment (blazoncek) [inline usermods/smartnest/usermod_smartnest.h:178]: Serial is initialised in `wled.cpp`there is no need to do it here.

File: usermods/smartnest/usermod_smartnest.h [hunk 2]
```diff
@@ -168,4 +168,40 @@ class Smartnest : public Usermod
...
+      Serial.println("Smartnest usermod setup initializing...");
```
Comment (blazoncek) [inline usermods/smartnest/usermod_smartnest.h:179]: Use `DEBUG_PRINTxx()` macros instead of `Serial.print()`.

File: usermods/smartnest/usermod_smartnest.h [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 #pragma once
 
 #include "wled.h"
 
 class Smartnest : public Usermod
 {
 private:
+  bool initialized = false;
+  unsigned long lastMqttReport = 0;
+  const unsigned long mqttReportInterval = 60000; // Report every minute
```
Comment (blazoncek) [inline usermods/smartnest/usermod_smartnest.h:10]: There  is really no need for this to be `unsigned long` and being `const` prevents this to be user adjustable via configuration.

File: usermods/smartnest/usermod_smartnest.h [hunk 4]
```diff
@@ -168,4 +168,40 @@ class Smartnest : public Usermod
...
+   * loop() is called continuously to keep the usermod running.
+   */
+  void loop() {
+    // Periodically report status to MQTT broker
+    unsigned long currentMillis = millis();
+    if (currentMillis - lastMqttReport >= mqttReportInterval) {
+      lastMqttReport = currentMillis;
+      
+      // Report current brightness
+      char brightnessMsg[4];
```
Comment (blazoncek) [inline usermods/smartnest/usermod_smartnest.h:198]: This assumes that `bri` will always be `uint8_t` which it is not. It may change to `uint16_t` in the future.

File: usermods/smartnest/usermod_smartnest.h [hunk 5]
```diff
@@ -1,14 +1,14 @@
-#ifndef WLED_ENABLE_MQTT
```
Comment (blazoncek) [inline usermods/smartnest/usermod_smartnest.h:1]: This should remain in IMO.

**Review by blazoncek — CHANGES_REQUESTED** [review]
There are a few things I'd like you to fix.

**Comment by blazoncek** [comment]
As I do not own Smartnest I'll tag @Steven17D as he is original author of the usermod to review and possibly test changes.

File: wled00/usermods_list.cpp [hunk 6]
```diff
@@ -9,7 +9,7 @@
  * || || ||
  * \/ \/ \/
  */
-//#include "../usermods/EXAMPLE_v2/usermod_v2_example.h"
```
Comment (blazoncek) [inline wled00/usermods_list.cpp:12]: Why did you remove the example?

**Review by blazoncek — CHANGES_REQUESTED** [review]
Adding a new usermod would require a new PR. Please revert changes and open a new PR for new usermod.
I also didn't notice that PR is based on `main` branch, please re-base to `0_15` branch.

File: wled00/usermods_list.cpp (hunk 6 above)
Comment (DevilPro1) [reply wled00/usermods_list.cpp:12]: Sorry, I was doing some tests on my own, I didn't know what it affected here
